src/server/transport.rs

```diff
@@ -44,12 +49,20 @@ where
 {
     fn send(&mut self, msg: RaftMessage) -> RaftStoreResult<()> {
         match self.raft_client.send(msg) {
-            Ok(()) => Ok(()),
+            Ok(()) => {
+                self.need_flush = true;
+                Ok(())
+            }
             Err(reason) => Err(raftstore::Error::Transport(reason)),
         }
     }
 
+    fn need_flush(&self) -> bool {
+        self.need_flush
+    }
+
     fn flush(&mut self) {
+        self.need_flush = false;
```

Comment:
May need to check the flag before flushing.

Reply:
I remove this line. You can see the new commit.